packages/api/src/utils/db-client.js

```diff
@@ -10,6 +11,165 @@ export class DBClient {
         Accept: '*/*',
       },
     })
+    this._clientNftStorage = new PostgrestClient(endpoint, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        Accept: '*/*',
+      },
+      schema: 'nftstorage',
+    })
+  }
+
+  /**
+   * @param {Object} permaCache
+   * @param {number} permaCache.userId
+   * @param {string} permaCache.url
+   * @param {number} permaCache.size
+   * @param {string} permaCache.insertedAt
+   */
+  async createPermaCache(permaCache) {
+    const { data, error, status } = await this._client
+      .from('perma_cache')
+      .insert({
+        user_id: permaCache.userId,
+        url: permaCache.url,
+        size: permaCache.size,
+        inserted_at: permaCache.insertedAt,
+      })
+      .single()
+
+    if (error) {
+      if (status === HTTP_STATUS_CONFLICT) {
+        throw new ConstraintError({
+          message: 'URL already found for user',
+          status: HTTP_STATUS_CONFLICT,
+        })
+      }
+      throw new DBError(error)
+    }
+
+    if (!data) {
+      throw new Error('Perma cache not created.')
+    }
+
+    return data
+  }
+
+  /**
+   * Get perma cache Entry
+   *
+   * @param {number} userId
+   * @param {string} url
+   * @return {Promise<{ url: string, size: number }>}
+   */
+  async getPermaCache(userId, url) {
+    const { data, status, error } = await this._client
+      .from('perma_cache')
+      .select(
+        `
+      url,
+      size,
+      insertedAt:inserted_at
+      `
+      )
+      .eq('user_id', userId)
+      .eq('url', url)
+      .is('deleted_at', null)
+      .single()
+
+    if (status === 406 || !data) {
+      return
+    }
+    if (error) {
+      throw new DBError(error)
+    }
+
+    return data
```

Comment:
406? Please make constant.

IMHO `.single()` is a footgun. It causes an error when results set contains 0 or >1 and we have to distinguish this error from actual transport errors. I find it much simpler to fetch the rows and return `rows[0]` - which will give you back a value or `undefined`, which is consistent with built in JS objects like `Map`/`Set` etc.

```suggestion
    if (error) {
      throw new DBError(error)
    }

    return data[0]
```